### PR TITLE
Misc tracking improvements and cleanup

### PR DESCRIPTION
**Pull request by jacobmcnamee**

### CODE REVIEW

**Comment by jacobmcnamee** [comment]
/cc @gsmcmullin 
*I promise I didn't change the behavior of anything*

File: src/board/v3/nap/track_channel.c [hunk 1]
```diff
@@ -28,74 +29,105 @@
...
+  t->CODE_INIT_G2 = 0x3ff;
+
+  t->SPACING = (SPACING_HALF_CHIP << NAP_TRK_SPACING_OUTER_Pos) |
+               (SPACING_HALF_CHIP << NAP_TRK_SPACING_INNER_Pos);
+
+  double cp_rate = (1.0 + carrier_freq / GPS_L1_HZ) * GPS_CA_CHIPPING_RATE;
+  nap_track_update(channel, carrier_freq, cp_rate, 0, 0);
 
-  NAP->TRK_CH[channel].SPACING = (SPACING_HALF_CHIP << 16) | SPACING_HALF_CHIP;
+  /* First integration is one sample short */
```
Comment (jacobmcnamee) [inline src/board/v3/nap/track_channel.c]: This was the reason for `calc_length_samples(1, 0, cp_rate)+1`, right?

Comment (gsmcmullin) [inline src/board/v3/nap/track_channel.c]: Yes, but I don't think this is very important now that the phases are read back from the NAP.

**Comment by gsmcmullin** [comment]
Looks like a good improvement.  Merge at will.

**Pull request merged**